sui-execution/src/latest.rs

```diff
@@ -71,7 +72,7 @@ impl executor::Executor for Executor {
         protocol_config: &ProtocolConfig,
         metrics: Arc<LimitsMetrics>,
         enable_expensive_checks: bool,
-        certificate_deny_set: &HashSet<TransactionDigest>,
+        early_execution_error: Option<ExecutionErrorKind>,
```

Comment:
I wonder if we want some sort of wider Result here. What I mean by that is, some of these fields are superfluous in the case that we are in an error, if I am understanding? So maybe we could/should try to encode that in the signature of the function as to avoid any sort of accidental execution in the presence of the error.